app/src/pages/review/[resourceType]/[id].tsx

```diff
@@ -1,30 +1,147 @@
+      if (JSON.stringify(resource?.extension) !== undefined) {
+        notifications.show({
+          title: 'Comment Successfully added!',
+          message: `Comment Successfully added to ${resourceType}/${resourceID}`,
+          icon: <CircleCheck />,
+          color: 'green'
+        });
+      }
+      ctx.draft.getDraftById.invalidate();
+    },
+    onError: e => {
+      notifications.show({
+        title: 'Update Failed!',
+        message: `Attempt to update ${resourceType} failed with message: ${e.message}`,
+        icon: <AlertCircle />,
+        color: 'red'
+      });
+    }
+  });
 
-  if (authoring === 'true') {
-    resource = trpc.draft.getDraftById.useQuery({
-      id: resourceID as string,
-      resourceType: resourceType as ArtifactResourceType
-    });
-  } else {
-    resource = trpc.service.getArtifactById.useQuery({
-      id: resourceID as string,
-      resourceType: resourceType as ArtifactResourceType
-    });
+  function getResource() {
+    if (authoring === 'true') {
+      return trpc.draft.getDraftById.useQuery({
+        id: resourceID as string,
+        resourceType: resourceType as ArtifactResourceType
+      });
+    } else {
+      return trpc.service.getArtifactById.useQuery({
+        id: resourceID as string,
+        resourceType: resourceType as ArtifactResourceType
+      });
+    }
   }
 
+  function parseUpdate(comment: string, type: string, userName: string, dateSelected: boolean) {
+    const additions: DraftArtifactUpdates = {};
+    const deletions: DraftArtifactUpdates = {};
+
+    const newExtensionObject: fhir4.Extension[] = [];
+    newExtensionObject.push({ url: 'type', valueCode: type }, { url: 'text', valueMarkdown: comment });
+
+    if (userName !== '') {
+      newExtensionObject.push({ url: 'user', valueString: userName });
+    }
+    if (dateSelected === true) {
+      const now = new Date();
+      const isoString = now.toISOString();
+      newExtensionObject.push({ url: 'authoredOn', valueDateTime: isoString });
+    }
+    if (resource?.extension) {
+      resource.extension.push({
+        extension: newExtensionObject,
+        url: 'http://hl7.org/fhir/us/cqfmeasures/CodeSystem/artifact-comment-type'
+      });
+      additions['extension'] = resource.extension;
+    }
+    return [additions, deletions];
+  }
+
+  // useEffect to check if the artifact has an extension and adds it if it doesn't
+  /* eslint-disable react-hooks/exhaustive-deps */
+  useEffect(() => {
+    if (!resource?.extension) {
+      const additions: DraftArtifactUpdates = {};
+      const deletions: DraftArtifactUpdates = {};
+      additions['extension'] = [];
+      resourceUpdate.mutate({
+        resourceType: resourceType as ArtifactResourceType,
+        id: resourceID as string,
+        additions: additions,
+        deletions: deletions
+      });
+    }
+  }, [resource]);
+  /* eslint-enable react-hooks/exhaustive-deps */
+
```

Comment:
We talked in person about this, but just putting this here for documentation purposes. I think we can remove the `useEffect` and do the check for is `resource.extension` exists in the `parseUpdate` function. I believe that this will not only be less code, but will also solve the weirdness we are seeing with `extension` being set to empty array.